pysoundfile_binaries/__init__.py

```diff
@@ -0,0 +1,131 @@
+sf_count_t  sf_readf_int     (SNDFILE *sndfile, int *ptr, sf_count_t frames) ;
+sf_count_t  sf_readf_float   (SNDFILE *sndfile, float *ptr, sf_count_t frames) ;
+sf_count_t  sf_readf_double  (SNDFILE *sndfile, double *ptr, sf_count_t frames) ;
+
+sf_count_t  sf_write_short   (SNDFILE *sndfile, short *ptr, sf_count_t items) ;
+sf_count_t  sf_write_int     (SNDFILE *sndfile, int *ptr, sf_count_t items) ;
+sf_count_t  sf_write_float   (SNDFILE *sndfile, float *ptr, sf_count_t items) ;
+sf_count_t  sf_write_double  (SNDFILE *sndfile, double *ptr, sf_count_t items) ;
+
+sf_count_t  sf_writef_short  (SNDFILE *sndfile, short *ptr, sf_count_t frames) ;
+sf_count_t  sf_writef_int    (SNDFILE *sndfile, int *ptr, sf_count_t frames) ;
+sf_count_t  sf_writef_float  (SNDFILE *sndfile, float *ptr, sf_count_t frames) ;
+sf_count_t  sf_writef_double (SNDFILE *sndfile, double *ptr, sf_count_t frames) ;
+
+sf_count_t  sf_read_raw      (SNDFILE *sndfile, void *ptr, sf_count_t bytes) ;
+sf_count_t  sf_write_raw     (SNDFILE *sndfile, void *ptr, sf_count_t bytes) ;
+
+const char* sf_get_string    (SNDFILE *sndfile, int str_type) ;
+int         sf_set_string    (SNDFILE *sndfile, int str_type, const char* str) ;
+
+typedef sf_count_t  (*sf_vio_get_filelen) (void *user_data) ;
+typedef sf_count_t  (*sf_vio_seek)        (sf_count_t offset, int whence, void *user_data) ;
+typedef sf_count_t  (*sf_vio_read)        (void *ptr, sf_count_t count, void *user_data) ;
+typedef sf_count_t  (*sf_vio_write)       (const void *ptr, sf_count_t count, void *user_data) ;
+typedef sf_count_t  (*sf_vio_tell)        (void *user_data) ;
+
+typedef struct SF_VIRTUAL_IO
+{    sf_count_t  (*get_filelen) (void *user_data) ;
+     sf_count_t  (*seek)        (sf_count_t offset, int whence, void *user_data) ;
+     sf_count_t  (*read)        (void *ptr, sf_count_t count, void *user_data) ;
+     sf_count_t  (*write)       (const void *ptr, sf_count_t count, void *user_data) ;
+     sf_count_t  (*tell)        (void *user_data) ;
+} SF_VIRTUAL_IO ;
+
+SNDFILE*    sf_open_virtual   (SF_VIRTUAL_IO *sfvirtual, int mode, SF_INFO *sfinfo, void *user_data) ;
+SNDFILE*    sf_open_fd        (int fd, int mode, SF_INFO *sfinfo, int close_desc) ;
+
+typedef struct SF_FORMAT_INFO
+{
+    int         format ;
+    const char* name ;
+    const char* extension ;
+} SF_FORMAT_INFO ;
+""")
+
+
+try:
+    snd = ffi.dlopen('sndfile')
+except OSError as err:
+    from sys import platform
+    from platform import architecture
+    from os.path import join, dirname
+
+    if platform == 'win32' and architecture()[0] == '32bit':
+        snd = ffi.dlopen(join(dirname(__file__), 'win32/libsndfile'))
+    elif platform == 'win32' and architecture()[0] == '64bit':
+        snd = ffi.dlopen(join(dirname(__file__), 'win64/libsndfile'))
+    elif platform == 'darwin':
+        snd = ffi.dlopen(join(dirname(__file__), 'darwin/libsndfile'))
+    else:
+        raise err
```

Comment:
You should just use `raise`.
Otherwise, you're losing the original traceback information.